packages/cli/src/controllers/project.controller.ts

```diff
@@ -210,44 +216,108 @@ export class ProjectController {
 	@Patch('/:projectId')
 	@ProjectScope('project:update')
 	async updateProject(
-		req: AuthenticatedRequest,
+		_req: AuthenticatedRequest,
 		_res: Response,
 		@Body payload: UpdateProjectDto,
 		@Param('projectId') projectId: string,
 	) {
-		const { name, icon, relations, description } = payload;
+		const { name, icon, description } = payload;
+		// Only update scalar settings on this endpoint. Member updates are handled by dedicated endpoints.
 		if (name || icon || description) {
 			await this.projectsService.updateProject(projectId, { name, icon, description });
 		}
-		if (relations) {
-			try {
-				const { project, newRelations } = await this.projectsService.syncProjectRelations(
-					projectId,
-					relations,
-				);
+	}
 
-				// Send email notifications to new sharees
+	/** Add one or more users to a project immediately */
+	@Post('/:projectId/users')
+	@ProjectScope('project:update')
+	async addProjectUsers(
+		req: AuthenticatedRequest,
+		res: Response,
+		@Param('projectId') projectId: string,
+		@Body payload: AddUsersToProjectDto,
+	) {
+		try {
+			const { added, conflicts, project } =
+				await this.projectsService.addUsersWithConflictSemantics(projectId, payload.relations);
+
+			if (added.length > 0) {
 				await this.userManagementMailer.notifyProjectShared({
 					sharer: req.user,
-					newSharees: newRelations,
+					newSharees: added,
 					project: { id: project.id, name: project.name },
 				});
-			} catch (e) {
-				if (e instanceof UnlicensedProjectRoleError) {
-					throw new BadRequestError(e.message);
-				}
-				throw e;
 			}
 
+			const relations = await this.projectsService.getProjectRelations(projectId);
+			this.eventService.emit('team-project-updated', {
+				userId: req.user.id,
+				role: req.user.role.slug,
+				members: relations.map((r) => ({ userId: r.userId, role: r.role.slug })),
+				projectId,
+			});
+
+			if (added.length > 0) return res.status(201).send();
+			if (conflicts.length > 0) return res.status(409).json({ conflicts });
```

Comment:
If I see that correctly we still insert some relations even if there are conflicts, so if we inserted some users in the line above, but there were also conflicts, we hide the conflicts correct?